core/src/scene/styleParam.cpp

```diff
@@ -592,106 +545,125 @@ bool StyleParam::parseTime(const std::string &_value, float &_time) {
 }
 
 template<typename T>
-int parseVec(const std::string& _value, T& _vec) {
-
-    const size_t elements = _vec.length();
-    const char* str = _value.data();
-
-    const int length = _value.length();
-    int count = 0;
-    int offset = 0;
-
-    for (size_t i = 0; i < elements; i++) {
-
-        float v = ff::stof(str + offset, length - offset, &count);
-        if (count == 0) { return i; }
-
-        _vec[i] = v;
-        offset += count;
-
-        if (length - offset <= 0) { return i; }
+bool parseVec(const YAML::Node& node, UnitSet allowedUnits, UnitVec<T>& vec) {
+    if (!node.IsSequence()) {
+        return false;
+    }
+    size_t nodeSize = node.size();
+    bool success = true;
+    for (size_t i = 0; i < vec.size; i++) {
+        if (i >= nodeSize) {
+            success = false;
+            break;
+        }
+        const auto& nodeElement = node[i];
+        if (!nodeElement.IsScalar()) {
+            success = false;
+            continue;
+        }
+        StyleParam::ValueUnitPair result;
+        success |= StyleParam::parseValueUnitPair(nodeElement.Scalar(), result);
```

Comment:
I am not sure if this logic is correct. If for one of the values in the sequence, `parseValueUnitPair` returns false, `success` being set to `true` above, will not be aptly set to `false`. Also probably it makes sense to `continue` if `parseValueUnitPair` for a value in the sequence returns false.

Reply:
@matteblair can you also add couple of unit tests to cover this logic. These things can be nasty to debug if broken, which unit tests can help avoid.